### PR TITLE
🎨: Update our symbol parsing code to extract additional function types

### PR DESCRIPTION
**Pull request by wldcordeiro**
Associated Issue: #3239 

### Summary of Changes

* enable additional parser features
* parse out some additional function types

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] Select a file that contains [class method arrow functions](https://github.com/devtools-html/debugger.html/compare/master...wldcordeiro:parse-more-babel?expand=1#diff-86a6158567e4b111270062b39add918eR10)
- [x] <kbd>cmd</kbd>+<kbd>shift</kbd>+<kbd>o</kbd> to open quick open modal to functions
- [x] See additional functions in list of results

### CODE REVIEW

**Comment by jasonLaster** [comment]
ooh... looks cool. I see a bunch of new omissions in the snapshots, can you document why that's the case?

File: src/workers/parser/tests/__snapshots__/closest.spec.js.snap [hunk 1]
```diff
@@ -93,12 +92,9 @@ FunctionDeclaration (9:0,11:1)
   body: BlockStatement (9:25,11:1)
     body:
       - ReturnStatement (10:2,10:15)
-          argument: StringLiteral (10:9,10:14)
-            extra:
-              raw: '"meh"'
-              rawValue: 'meh'
+          argument: Literal (10:9,10:14)
```
Comment (wldcordeiro) [inline src/workers/parser/tests/__snapshots__/closest.spec.js.snap]: Not sure why the changes here happened bu they seem like they don't matter too much.

File: src/workers/parser/tests/__snapshots__/getEmptyLines.spec.js.snap [hunk 2]
```diff
@@ -15,9 +15,10 @@ Array [
 exports[`getEmptyLines class 1`] = `
 Array [
   5,
-  10,
-  12,
+  9,
```
Comment (wldcordeiro) [inline src/workers/parser/tests/__snapshots__/getEmptyLines.spec.js.snap]: Lines changed because I added to the file.

**Comment by wldcordeiro** [comment]
Marking WIP for now while I investigate the regressions.

File: src/workers/parser/utils/ast.js [hunk 3]
```diff
@@ -14,7 +14,20 @@ function _parse(code, opts) {
     code,
     Object.assign({}, opts, {
       sourceType: "module",
-      plugins: ["jsx", "flow", "objectRestSpread"]
+      plugins: [
+        "estree",
```
Comment (wldcordeiro) [inline src/workers/parser/utils/ast.js]: Turns out this plugin was the problem @jasonLaster removing it reverted a lot of the snapshot changes.
